### PR TITLE
[NEXUS-8318] BASIC-auth challenge fixes

### PR DESCRIPTION
**Pull request by jdillon**
https://issues.sonatype.org/browse/NEXUS-8318

### CODE REVIEW

**Comment by jdillon** [comment]
http://bamboo.s/browse/NX3-OSSF123-1 :white_check_mark:

**Comment by kellyrob99** [comment]
+1 tested and working for the reported maven deploy issue

**Comment by cstamas** [comment]
+1

File: components/nexus-repository/src/main/java/org/sonatype/nexus/repository/security/SecurityFacet.java [hunk 1]
```diff
@@ -26,7 +27,9 @@
   extends Facet
 {
   /**
-   * Check if the given request is permitted on the the repository.
+   * Ensure the given request is permitted on the the repository.
```
Comment (adreghiciu) [inline components/nexus-repository/src/main/java/org/sonatype/nexus/repository/security/SecurityFacet.java]: spelling

**Comment by adreghiciu** [comment]
+1

**Comment by cstamas** [comment]
Locally tested this, and it does not work for me. Seems the Shiro exception is not let past [this point](https://github.com/sonatype/nexus-oss/blob/NEXUS-8318-basic-auth-challange/plugins/basic/nexus-repository-httpbridge/src/main/java/org/sonatype/nexus/repository/httpbridge/internal/ViewServlet.java#L117)

All I have is a nice stack trace like this one:
```
2015-03-24 18:34:52,546+0100 WARN  [qtp1462165892-51] *UNKNOWN org.sonatype.nexus.repository.httpbridge.internal.ViewServlet - Service failure
org.apache.shiro.authz.AuthorizationException: User is not permitted: [nexus]:[repository-view]:[maven2]:[maven2-group1]:[read]
	at org.sonatype.nexus.security.authz.ExceptionCatchingModularRealmAuthorizer.checkPermission(ExceptionCatchingModularRealmAuthorizer.java:75) [na:na]
	at org.sonatype.nexus.security.authz.ExceptionCatchingModularRealmAuthorizer.checkPermissions(ExceptionCatchingModularRealmAuthorizer.java:93) [na:na]
	at org.apache.shiro.mgt.AuthorizingSecurityManager.checkPermissions(AuthorizingSecurityManager.java:149) [na:na]
	at org.apache.shiro.subject.support.DelegatingSubject.checkPermissions(DelegatingSubject.java:220) [org.apache.shiro.core:1.2.3]
	at org.sonatype.nexus.security.SecurityHelper.ensurePermitted(SecurityHelper.java:71) [na:na]
	at org.sonatype.nexus.security.SecurityHelper.ensurePermitted(SecurityHelper.java:78) [na:na]
	at org.sonatype.nexus.repository.security.SecurityFacetSupport.ensurePermitted(SecurityFacetSupport.java:63) [na:na]
	at org.sonatype.nexus.repository.security.SecurityHandler.handle(SecurityHandler.java:44) [na:na]
	at org.sonatype.nexus.repository.view.Context.proceed(Context.java:89) [na:na]
	at org.sonatype.nexus.repository.view.handlers.TimingHandler.handle(TimingHandler.java:46) [na:na]
	at org.sonatype.nexus.repository.view.Context.proceed(Context.java:89) [na:na]
	at org.sonatype.nexus.repository.view.Context.start(Context.java:112) [na:na]
	at org.sonatype.nexus.repository.view.Router.dispatch(Router.java:58) [na:na]
	at org.sonatype.nexus.repository.view.ConfigurableViewFacet.dispatch(ConfigurableViewFacet.java:55) [na:na]
	at org.sonatype.nexus.repository.httpbridge.internal.ViewServlet.dispatchAndSend(ViewServlet.java:165) [na:na]
	at org.sonatype.nexus.repository.httpbridge.internal.ViewServlet.doService(ViewServlet.java:152) [na:na]
	at org.sonatype.nexus.repository.httpbridge.internal.ViewServlet.service(ViewServlet.java:112) [na:na]
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:790) [javax.servlet-api:3.1.0]
...
```

**Comment by cstamas** [comment]
Was wrong: the challenge is sent, but the WARN log did confuse me.

**Comment by jdillon** [comment]
@cstamas I didn't spend time to reduce logging for expected exceptions, we have to generally clean all that up before 3.0 is final, but for now I left it as simple as possible

File: components/nexus-repository/src/main/java/org/sonatype/nexus/repository/security/SecurityFacet.java (hunk 1 above)
Comment (jdillon) [inline components/nexus-repository/src/main/java/org/sonatype/nexus/repository/security/SecurityFacet.java]: do you mean "the the", thats technically not a spelling problem :-P

**Comment by jdillon** [comment]
@cstamas shiro filters operate _around_ the servlet.  And there is specific handling in Shiro (and in my changes) to cope with plain Shiro exception as well as wrapped in ServletException, though ServletException wrapping is actually done by one of the Shiro filters, the propagate  if possible will propagate the AuthorizationException as-is since its a RuntimeException derivative.

**Comment by jdillon** [comment]
replaced by https://github.com/sonatype/nexus-oss/pull/1135 due to merge/push mess up

**Pull request closed without merging**